lstchain/scripts/lstchain_mc_dl1_to_dl2.py

```diff
@@ -47,6 +48,24 @@
                     default=None
                     )
 
+parser.add_argument('--source_dependent', '-srcdep', action='store', type=lambda x: bool(strtobool(x)),
+                    dest='src_dependent',
+                    help='Boolean. True for source-dependent analysis. Default = False (source-independent analysis)',
+                    default=False)
+
+parser.add_argument('--expected_src_pos_x', '-srcposx', action='store', type=str,
```

Comment:
at the moment, if I'm not wrong, the source position simulated in the MC is always taken as one point (for the point-like one). What will happen when instead of a point source we simulate a ring (same distance from the center of the camera, but a circle around it)? The expected source position is probably better read from the file since it will be different for every event (at the moment it is not, but just because of the characteristics of the simulation we are using right now)

Reply:
You're right. I modified to read expected source position from the file for MC data, only for gamma. For proton MC/real data dist is defined as just the distance from the center of the camera.
But for real data we need pointing information including wobble information(wobble offset/angle) to know the expected source position. I saw telescope alt/az information in drive report, but there was no information about the pointing offset if I didn't miss it.
So for the moment dist is defined only for ON observation mode, then I will modify later.
(At that time I will also modify the code following the above comments)